core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -813,13 +813,23 @@ private[chisel3] object Builder extends LazyLogging {
 
   // Helper for reasonable errors when clock or reset value not yet initialized
   private def getDelayed[A](field: String, dc: Delayed[A]): A = {
-    val result = dc.value
-    if (result == null) {
-      // TODO add SourceInfo and change to Builder.exception
+    // TODO add SourceInfo and change to Builder.exception
+    def err(extra: String) = {
       throwException(
-        s"The implicit $field is null which means the code that sets its definition has not yet executed."
+        s"The implicit $field is null which means the code that sets its definition has not yet executed. $extra."
       )
     }
+    val result =
+      try {
+        dc.value
+      } catch {
+        // If user uses -Xcheckinit there will be additional information
+        case UninitializedFieldError(msg) => err(msg)
+      }
+    // If user is not using -Xcheckinit, suggest that they do
+    if (result == null) {
+      err("Try adding -Xcheckinit to your scalacOptions to get a more useful stack trace")
+    }
     result
```

Comment:
This was needed to fix some tests that failed due to our now using `-Xcheckinit`. I made it such that the error suggestion works both with and without `-Xcheckinit` and either gives you the additional context from using that option or suggests you use `-Xcheckinit`.